src/main/java/blusunrize/immersiveengineering/common/IEContent.java

```diff
@@ -199,14 +199,17 @@ public static void init(ParallelDispatchEvent ev)
 		// Buckets
 		// TODO add "duplicates" of the fluid-aware recipes that only use buckets, so that other mods using similar
 		//  code don't need explicit compat?
-		AssemblerHandler.registerSpecialIngredientConverter((o) ->
+		AssemblerHandler.registerSpecialIngredientConverter((o, remain) ->
 		{
 			final ItemStack[] matching = o.getItems();
 			if(!o.isVanilla()||matching.length!=1)
 				return null;
 			final Item potentialBucket = matching[0].getItem();
 			if(!(potentialBucket instanceof BucketItem))
 				return null;
+			// bucket was consumed in recipe
+			if(remain.isEmpty())
```

Comment:
Might be better to explicitly check for an empty bucket here? Probably won't happen in practice, but just in case some mod returns something that isn't a bucket. Same in the milk check below.

Reply:
Fair